### PR TITLE
[mempool] Remove legacy failover logic

### PR DESCRIPTION
**Pull request by gregnazario**
Currently, we only use broadcast to all failovers, rather than logic to
choose a single failover.  So, I'm cutting out that failover logic in
addition to a test that is specific only to it.

### CODE REVIEW

File: mempool/src/tests/shared_mempool_test.rs [hunk 1]
```diff
@@ -779,166 +779,6 @@ fn test_vfn_multi_network() {
     smp.assert_no_message_sent(&vfn_0);
 }
 
-#[test]
-fn test_fn_failover() {
-    // Test vfn failing over to fallback network
```
Comment (davidiw) [inline mempool/src/tests/shared_mempool_test.rs:784]: 🥳 
we should still validate that if the upstream is down that the txn is committed, do we have a test for that?

Comment (gregnazario) [reply mempool/src/tests/shared_mempool_test.rs:784]: Yeah, working on replacing this current test with a new one, that tests with the other failover mode.

There is a different test that tests the "everything is running" case already.

**Review by davidiw — APPROVED** [review]
that's some interesting test code :O

**Pull request merged**